programs/vote/src/vote_state/mod.rs

```diff
@@ -727,26 +727,23 @@ pub fn process_vote(
     vote: &Vote,
     slot_hashes: &[SlotHash],
     epoch: Epoch,
-    feature_set: Option<&FeatureSet>,
+    filter_vote_slots: bool,
```

Comment:
I think it's fine here to completely remove this bool and just always filter. Even the one place where this boolean is false https://github.com/solana-labs/solana/blob/8264cff2f1153c07b4919e763649644df5db30ac/programs/vote/src/vote_state/mod.rs#L769 already inserted all the `SlotHashes`: https://github.com/solana-labs/solana/blob/8264cff2f1153c07b4919e763649644df5db30ac/programs/vote/src/vote_state/mod.rs#L763 from the votes it was handed so nothing would get filtered anyways

cc @AshwinSekar

Reply:
`process_vote_unchecked()` is used in Tower consensus.rs and I think it might also make sense to keep it as it is more performant without the filtering.

Reply:
yeah it's used a couple places in `consensus.rs` but still is creating the dummy `slot_hashes` for nothing. perhaps the best median would be to make `slot_hashes` an `Option` and only filter when it is `Some` rather than have this extra `bool`

Reply:
`check_slots_are_valid()` (which is called inside `process_vote()`) also takes `slot_hashes` so we can't make it optional, right?

Reply:
ah right, yeah i think it's fine as you have it now. 

for a future refactor i think `process_vote()` should perform the checks and then call `process_vote_unchecked()` rather than `process_vote_unchecked()` passing the dummy `slot_hashes` in order to skip the checks in `process_vote()`

Reply:
I think it might be cleaner to extract this section: https://github.com/solana-labs/solana/blob/8264cff2f1153c07b4919e763649644df5db30ac/programs/vote/src/vote_state/mod.rs#L748-L758 into a separate function `do_process_vote()`, called by both `process_vote` and `process_vote_unchecked()`. Then we can leave https://github.com/solana-labs/solana/blob/8264cff2f1153c07b4919e763649644df5db30ac/programs/vote/src/vote_state/mod.rs#L736-L742 in `process_vote` without introducing a new flag